adapter/src/main/kotlin/org/javacs/ktda/jdi/launch/KDACommandLineLauncher.kt

```diff
@@ -0,0 +1,208 @@
+package org.javacs.ktda.jdi.launch
+
+import com.sun.jdi.Bootstrap
+import com.sun.jdi.VirtualMachine
+import com.sun.jdi.connect.*
+import com.sun.jdi.connect.spi.Connection
+import com.sun.jdi.connect.spi.TransportService
+import org.codehaus.plexus.util.cli.CommandLineUtils
+import org.javacs.kt.LOG
+import java.io.File
+import java.io.IOException
+import java.net.URLDecoder
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.util.*
+import java.util.concurrent.Callable
+import java.util.concurrent.Executors
+import java.util.concurrent.TimeUnit
+
+internal const val ARG_HOME = "home"
+internal const val ARG_OPTIONS = "options"
+internal const val ARG_MAIN = "main"
+internal const val ARG_SUSPEND = "suspend"
+internal const val ARG_QUOTE = "quote"
+internal const val ARG_VM_EXEC = "vmexec"
+internal const val ARG_CWD = "cwd"
+internal const val ARG_ENVS = "envs"
+
+/**
+ * A custom LaunchingConnector that supports cwd and env variables
+ */
+open class KDACommandLineLauncher : LaunchingConnector {
+
+    protected val defaultArguments = mutableMapOf<String, Connector.Argument>()
+
+    /**
+     * We only support SocketTransportService
+     */
+    protected var transportService : TransportService? = null
+    protected var transport = Transport { "dt_socket" }
+
+    companion object {
+
+        fun urlEncode(arg: Collection<String>?) = arg
+                ?.map { URLEncoder.encode(it, StandardCharsets.UTF_8.name()) }
+                ?.fold("") { a, b -> "$a\n$b" }
+
+        fun urlDecode(arg: String?) = arg
+                ?.trim('\n')
+                ?.split("\n")
+                ?.map { URLDecoder.decode(it, StandardCharsets.UTF_8.name()) }
+                ?.toList()
+    }
+
+    constructor() : super() {
+
+        defaultArguments[ARG_HOME] = StringArgument(ARG_HOME, description = "Java home", value = System.getProperty("java.home"))
+
+        defaultArguments[ARG_OPTIONS] = StringArgument(ARG_OPTIONS, description = "Jvm arguments")
+
+        defaultArguments[ARG_MAIN] = StringArgument(ARG_MAIN, description = "Main class name and parameters", mustSpecify = true)
+
+        defaultArguments[ARG_SUSPEND] = StringArgument(ARG_SUSPEND, description = "Whether launch the debugee in suspend mode", value = "true")
+
+        defaultArguments[ARG_QUOTE] = StringArgument(ARG_QUOTE, description = "Quote char", value = "\"")
+
+        defaultArguments[ARG_VM_EXEC] = StringArgument(ARG_VM_EXEC, description = "The java exec", value = "java")
+
+        defaultArguments[ARG_CWD] = StringArgument(ARG_CWD, description = "Current working directory")
+
+        defaultArguments[ARG_ENVS] = StringArgument(ARG_ENVS, description = "Environment variables")
```

Comment:
Style nit: Please remove the blank lines inbetween.

```suggestion
        defaultArguments[ARG_HOME] = StringArgument(ARG_HOME, description = "Java home", value = System.getProperty("java.home"))
        defaultArguments[ARG_OPTIONS] = StringArgument(ARG_OPTIONS, description = "Jvm arguments")
        defaultArguments[ARG_MAIN] = StringArgument(ARG_MAIN, description = "Main class name and parameters", mustSpecify = true)
        defaultArguments[ARG_SUSPEND] = StringArgument(ARG_SUSPEND, description = "Whether launch the debugee in suspend mode", value = "true")
        defaultArguments[ARG_QUOTE] = StringArgument(ARG_QUOTE, description = "Quote char", value = "\"")
        defaultArguments[ARG_VM_EXEC] = StringArgument(ARG_VM_EXEC, description = "The java exec", value = "java")
        defaultArguments[ARG_CWD] = StringArgument(ARG_CWD, description = "Current working directory")
        defaultArguments[ARG_ENVS] = StringArgument(ARG_ENVS, description = "Environment variables")
```